### PR TITLE
Added snippets for SupportingPaneScaffold

### PR DESCRIPTION
**Pull request by IanGClifton**

### CODE REVIEW

File: compose/snippets/src/main/java/com/example/compose/snippets/adaptivelayouts/SampleSupportingPaneScaffold.kt [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        navigator.navigateBack()
+    }
+
+    SupportingPaneScaffold(
+        directive = navigator.scaffoldDirective,
+        value = navigator.scaffoldValue,
+        mainPane = {
+            AnimatedPane(modifier = Modifier.safeContentPadding()) {
+                // Main pane content
+                if (navigator.scaffoldValue.secondary == PaneAdaptedValue.Hidden) {
```
Comment (alexvanyo) [inline compose/snippets/src/main/java/com/example/compose/snippets/adaptivelayouts/SampleSupportingPaneScaffold.kt]: The `get()` overload here looks a bit nicer in my eyes:

```suggestion
                if (navigator.scaffoldValue[SupportingPaneScaffoldRole.Supporting] == PaneAdaptedValue.Hidden) {
```

File: compose/snippets/src/main/java/com/example/compose/snippets/adaptivelayouts/SampleSupportingPaneScaffold.kt [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        },
+    )
+    // [END android_compose_adaptivelayouts_sample_supporting_pane_scaffold_full]
+}
+
+// [START android_compose_adaptivelayouts_sample_supporting_pane_scaffold_extracted_panes]
+@Composable
+fun ThreePaneScaffoldScope.MainPane(
+    shouldShowSupportingPaneButton: Boolean,
+    onNavigateToSupportingPane: () -> Unit,
```
Comment (alexvanyo) [inline compose/snippets/src/main/java/com/example/compose/snippets/adaptivelayouts/SampleSupportingPaneScaffold.kt:100]: Should this have a `modifier: Modifier = Modifier` for completeness?

File: compose/snippets/src/main/java/com/example/compose/snippets/adaptivelayouts/SampleSupportingPaneScaffold.kt [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+                Text("Show supporting pane")
+            }
+        } else {
+            Text("Supporting pane is shown")
+        }
+    }
+}
+
+@Composable
+fun ThreePaneScaffoldScope.SupportingPane() {
```
Comment (alexvanyo) [inline compose/snippets/src/main/java/com/example/compose/snippets/adaptivelayouts/SampleSupportingPaneScaffold.kt]: Ditto here for `modifier`

File: compose/snippets/src/main/java/com/example/compose/snippets/adaptivelayouts/SampleSupportingPaneScaffold.kt (hunk 2 above)
Comment (IanGClifton) [reply compose/snippets/src/main/java/com/example/compose/snippets/adaptivelayouts/SampleSupportingPaneScaffold.kt:100]: Good call, yeah let me do that to reinforce best practices

**Review by alexvanyo — APPROVED** [review]

**Pull request merged**